streams/src/main/java/org/apache/kafka/streams/kstream/internals/ConsumedInternal.java

```diff
@@ -42,10 +43,18 @@ public ConsumedInternal() {
         return keySerde;
     }
 
+    public Deserializer<K> keyDeserializer() {
+        return keySerde == null ? null : keySerde.deserializer();
+    }
+
     public Serde<V> valueSerde() {
         return valueSerde;
     }
 
+    public Deserializer<V> valueDeserializer() {
+        return valueSerde == null ? null : valueSerde.deserializer();
+    }
+
```

Comment:
cc @dguy

Reply:
I think this looks good. But left for @dguy to make a call.

Reply:
yep LGTM